docker/scripts/entrypoint.sh

```diff
@@ -37,10 +37,13 @@ export BK_journalDirectory=${BK_journalDirectory:-${BK_DATA_DIR}/journal}
 export BK_ledgerDirectories=${BK_ledgerDirectories:-${BK_DATA_DIR}/ledgers}
 export BK_indexDirectories=${BK_indexDirectories:-${BK_DATA_DIR}/index}
 
+export BK_statsProviderClass=org.apache.bookkeeper.stats.NullStatsProvider
```

Comment:
This should be
```
export BK_statsProviderClass=${BK_statsProviderClass:-org.apache.bookkeeper.stats.NullStatsProvider}
```

Otherwise, the user can never override this.

Reply:
@ivankelly I think you misunderstood what the change is doing.

I believe that the change is here the correct behavior. If I understand correctly, in 4.5.1, none of the stats providers are packaged into the release binary and docker image. setting this to `NullStatsProvider` is to avoid people overriding this to other stats providers, and cause `ClassNotFoundException` exception.

@eolivelli changed that in the makefile only #746 , which only applies to dockers run by the Makefile. but that change doesn't prevent the error when people use the docker image in other environment, e.g. gke. I think the change here is overriding this stats provider and avoid user trying to customize other stats providers. 

I also think this change is only need in 4.5, but it isn't needed in 4.6 and onwards. because 4.6 switch to bookkeeper-dist for release package and docker image, it will package the stats providers in the package.

Reply:
@sijie Thanks for the comments, it is only for 4.5.1.  and sorry for the un-complete change in 746.

Reply:
@sijie @jiazhai 
Ah ok, PR description should be updated to state that then. Why have a variable for stats provider at all if it's going to be constant?

Do we have a 4.5.2 release scheduled?

Reply:
I think the PR is filed against 4.5 and it is labelled as `release/4.5.1`, no?

the variable is used the `docker/scripts/apply-config-from-env.py`. because the script will pick up environment variables for applying the settings to the config file.

this change is patching docker files. it doesn't require generating a new package. we just need to push this change, so the docker hub can automatically rebuild the image. so, no 4.5.2 release scheduled yet.

Reply:
@sijie I mean, the PR description should state that 4.5.1 doesn't ship the providers, so we hard code it to the null providers.

Reply:
oh okay.